client/src/components/ui/signup/pw.tsx

```diff
@@ -0,0 +1,177 @@
+import axios, { AxiosResponse } from "axios";
+import Image from "next/image";
+import { ChangeEvent, useState } from "react";
+
+import { Button } from "@/components/ui/button";
+import { ErrorCallout } from "@/components/ui/callout";
+import { HTMLTextTargetElement, SingleLineInput } from "@/components/ui/inputs";
+import api from "@/lib/api";
+
+import { pwRegex } from "./regex";
+
+//import { isCommonPassword } from "./check-common-pw";
+
+interface PWProps {
+  emailMobile: string;
+  currentStep: number;
+  password: string;
+  setPassword: React.Dispatch<React.SetStateAction<string>>;
+  setCurrentStep: React.Dispatch<React.SetStateAction<number>>;
+}
+
+function passwordValidation(password: string, repeatPassword: string): boolean {
+  return repeatPassword === password;
+}
+
+export const PW: React.FC<PWProps> = ({
+  currentStep,
+  setCurrentStep,
+  password,
+  setPassword,
+}) => {
+  const [passwordVariant, setPasswordVariant] = useState("inactive");
+  const [repeatPassword, setRepeatPassword] = useState("");
+  const [repeatPasswordVariant, setRepeatPasswordVariant] =
+    useState("inactive");
+  const [errorMessage, setErrorMessage] = useState<string | null>(null);
+
+  async function passwordCommonCheck(
+    password: string,
+  ): Promise<string | undefined> {
+    try {
+      const response = await api.post("/app/validate/", { password: password });
+
+      if (response.status === 200) {
+        console.log("Password is valid:", response.data.valid);
+      }
+    } catch (error) {
+      if (axios.isAxiosError(error)) {
+        console.error("Validation errors:", error.response?.data.errors);
+        return error.response?.data.errors;
+      } else {
+        console.error("Error:", error);
+      }
+    }
+  }
+
+  function inputCheck(
+    password: string,
+    hook: React.Dispatch<React.SetStateAction<string>>,
+  ) {
+    if (password.length <= 8) {
+      setErrorMessage(
+        "This password is too short. It must contain at least 8 characters.",
+      );
+      hook("inactive");
+      return false;
+    } else if (pwRegex.test(password)) {
+      setErrorMessage("This password is entirely numeric.");
+      hook("inactive");
+      return false;
+    } else {
+      setErrorMessage(null);
+      hook("default");
+      return true;
+    }
+  }
+
+  const handlePasswordChange = (e: ChangeEvent<HTMLTextTargetElement>) => {
+    const value = e.target.value;
+    setPassword(value);
+    setPasswordVariant(e.target.value ? "default" : "inactive");
+    inputCheck(value, setPasswordVariant);
+  };
+
+  const handleRepeatPasswordChange = (
```

Comment:
This one doesn't check when changing the first password input field.
<img width="1303" alt="image" src="https://github.com/user-attachments/assets/c0df3ddf-534b-474b-875a-9bfaf5c82410">
Needs to add to the first pw input field